binaries/daemon/src/spawn.rs

```diff
@@ -101,13 +100,10 @@ pub async fn spawn_node(
                 source => {
                     let resolved_path = if source_is_url(source) {
                         // try to download the shared library
-                        let target_path = Path::new("build")
-                            .join(node_id.to_string())
-                            .with_extension(EXE_EXTENSION);
-                        download_file(source, &target_path)
+                        let target_dir = Path::new("build");
+                        download_file(source, &target_dir)
                             .await
-                            .wrap_err("failed to download custom node")?;
-                        target_path.clone()
+                            .wrap_err("failed to download custom node")?
```

Comment:
Could you clarify the reason for this breaking change?

By renaming the downloaded files we avoided collisions. For example, consider:

```yaml
nodes:
  - id: camera
    path: example.com/camera/node
    inputs:
      tick: dora/timer/millis/20
    outputs:
      - image

  - id: plot
    path: example.com/plot/node
    inputs:
      image: camera/image
```

With the new logic, the second download overwrites the first, as both nodes are stored as `build/node`. It probably still works as expected on Linux because the executables are directly started, but I could imagine that there will be some errors on Windows (since Windows disallows removing executable files that are still running). Also, this collision would prevent us from adding some kind of caching again in the future.

Reply:
I understand you're reasoning. 

But, I really don't think we should rename file on the fly as it makes it very hard to keep track of which file correspond to what.

I think that if people name everything "node.py" and colludes with names, I don't think that changing every name is the right solution.

Reply:
I would very much prefer people use something like:

```bash
  - id: plot
    build: wget -O plot.py example.com/plot/node
    path: node.py
    inputs:
      image: camera/image
```

Reply:
> I think that if people name everything "node.py" and colludes with names, I don't think that changing every name is the right solution.

The thing is that the two `node.py` files might come from different sources. For example, they could be part of different git repositories that are not your own. In this case, you have no way to avoid the name conflict.

> But, I really don't think we should rename file on the fly as it makes it very hard to keep track of which file correspond to what.

For the above example, you don't have the source for one of the nodes at all (because it's overwritten).




> I would very much prefer people use build for caching, in the likes of something like:

Sure, this is a good alternative. But if we support pointing `path` to remote URLs, we should be able to deal with potential name collisions.

Reply:
> Sure, this is a good alternative. But if we support pointing path to remote URLs, we should be able to deal with potential name collisions.

Ok, so I guess, what we could do is fail on pre-existing files. That would make the most sense to me.

Reply:
The other alternative would be renaming, either always or only on conflict. If we don't want to do that, I agree that failing is better than silently overwriting the files of other nodes.